jakarta.ls/src/main/java/io/microshed/jakartals/JakartaTextDocumentService.java

```diff
@@ -78,6 +84,20 @@ public void didClose(DidCloseTextDocumentParams params) {
 				.publishDiagnostics(new PublishDiagnosticsParams(uri, new ArrayList<Diagnostic>()));
 	}
 
+	@Override
+	public CompletableFuture<Either<List<CompletionItem>, CompletionList>> completion(CompletionParams position) {
+		/*
+		Code completion functionality for eclipse Jakarta EE.
+		This method is automatically called by the Language Server Client
+		provided it has provided a java-completion-computer extension on the client side.
+		*/
+		// Find a way to maintain a context Per Document
+		// We then pass this context into the CompletionItem with what the current URI has in its classpath
```

Comment:
is this a todo?  Is the idea that the context will offer us insight into the classpath?

Reply:
Yeah sorry that is a todo, I will remove it from the PR, just to keep it clean.

Reply:
And yeah maintaining information about each document such as imports and such will allow us to perform direct compares with the snippet context to determine if it should be shown or not. That is how I think it'll work so far, but I will have a more solid answer once I actually start looking into context based snippet displaying

Reply:
Sounds good, thanks Ankush :)